### DISCUSSION TITLE
Is it possible to perform a DMFT calculation of a ferromagnetic material with Spin-Orbit Coupling(SOC)  considered?

### DISCUSSION THREAD

**Discussion by maplezmf** [TRIQS/solid_dmft#42] [Q&A] [answered]
Dear TRIQS/solid_dmft community,
  As a newbee of DMFT, I find in Quantum Espresso v7.0(PW/src/input.f90 line1250) that 'DMFT update not implemented with nspin > 1'. Is it mean that one can not perform a DMFT calculation of a ferromagnetic material with SOC with QE? 
If yes, is there any other packages/software that can perform such calculations, such as VASP v5.4.1?  As  I also find in DFTtools, even with Wien2k, as in tuturial: Sr2MgOsO6 with SOC (non-magnetic), can perform dmft calculations of a non-magnetic system.
   
sincerely yours 
Maplezmf

**Comment by the-hampel** [comment, marked as answer]
Hi @maplezmf,
Typically the spin symmetry is lifted, i.e. magnetic calculations is performed, only on the DMFT level. This is why there are no tutorials in TRIQS or solid_dmft performing also the DFT calculation magnetic. For example in this tutorial we show how to perform a magnetic calculation with Vasp + solid_dmft: https://triqs.github.io/solid_dmft/tutorials/NNO_os_plo_mag/tutorial.html . The reason is that using the spin polarized exchange correlation functionals induces a large change how exchange / correlations are treated and double counting becomes highly non trivial when performing DMFT calculations on top.  Hence, there is a check that the user is not trying to perform also the DFT calculation with spin. 

However, this does not prevent you from doing magnetic DFT+DMFT calculations, even charge self-consistency is not a problem. But for the QE interface this is not implemented in dfttools right now. Vasp averages up and down channels by default. Wien2k and Elk allow you to read in updates for up and down separately. However, one should be careful with this feature. The limiting factor is this function https://triqs.github.io/dft_tools/unstable/_python_api/triqs_dft_tools.sumk_dft.SumkDFT.calc_density_correction.html in triqs/dft_tools. Not solid_dmft.

To summarize: if you like to perform a simple one shot DMFT calculation with spin and SOC any of the codes should work fine (there is a question on how to construct an interaction Hamiltonian in these cases but this is a different thing). Charge self-consistent updates with spin might be a bit more tricky. They should work in some form with Vasp (SOC only on unstable branch of dft tools), Elk, and Wien2k. Not with QE as of now.

Best,
Alex

**Reply by maplezmf** [reply]
Thank you very much for your swift reply. That helps a lot!
